frontend/src/pages/CompareV2.tsx

```diff
@@ -168,6 +424,28 @@ function CompareV2(props: PageProps) {
   const [isParamsCollapsed, setIsParamsCollapsed] = useState(false);
   const [isMetricsCollapsed, setIsMetricsCollapsed] = useState(false);
 
+  const [confusionMatrixArtifacts, setConfusionMatrixArtifacts] = useState<RunArtifact[]>([]);
+  const [htmlArtifacts, setHtmlArtifacts] = useState<RunArtifact[]>([]);
+  const [markdownArtifacts, setMarkdownArtifacts] = useState<RunArtifact[]>([]);
+
+  // Selected artifacts for two-panel layout.
+  const createSelectedArtifactArray = (count: number): SelectedArtifact[] => {
+    const array: SelectedArtifact[] = [];
+    for (let i = 0; i < count; i++) {
+      array.push({
+        selectedItem: { itemName: '', subItemName: '' },
+      });
+    }
+    return array;
+  };
```

Comment:
This method is used so each `SelectedArtifact` object does not share the same reference, as updating one would then update the rest.